### PR TITLE
Add sqlite3 gem

### PR DESCRIPTION
**Pull request by karlhungus**
For vagrant users:

sudo apt-get install sqlite3
sudo apt-get install libsqlite3-dev
bundle install

### CODE REVIEW

**Comment by karlhungus** [comment]
@ibawt not sure about the db port in this one, didn't have direct push access to `ibawt/eightfoot`, don't mind working this way though.

Not positive I got my client app/secret right but couldn't login to github, not sure if it's a vagrant thing or what.

**Comment by ibawt** [comment]
i had sqllite in there before, but it won't let me do migrations where i drop a column :S.  Erm oh I should have deleted it from the gemfile.

the mysql port is our old boxen one.  

I haven't logged in a bit, but I think @qq99 logged in successfully.  

I'll add you to the team though.

File: config/database.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ development:
   pool: 5
   username: root
   password:
-  port: 13306
+  port: 3306
```
Comment (ibawt) [inline config/database.yml]: should do the if BOXEN thing we have for shopify in there

Comment (karlhungus) [inline config/database.yml]: makes more sense, i'll do this

**Comment by karlhungus** [comment]
We could just move the test database to mysql? or drop it manually for migrations? both solutions seem kind of craptacular

File: config/database.yml (hunk 1 above)
Comment (tylermercier) [inline config/database.yml]: This one?

```
<%= ENV.fetch('BOXEN_MYSQL_PORT', 3306) %>
```

An alternative would be to git-ignore database.yml and have the user copy database.example.yml

**Comment by karlhungus** [comment]
oh look, it's by design: http://www.sqlite.org/omitted.html.

**Comment by karlhungus** [comment]
sign in works fine (pebkac:didn't press the update button after updating )

File: config/database.yml (hunk 1 above)
Comment (qq99) [inline config/database.yml]: Yep, I had to change this to the non-boxen port, might be a few other such config (not sure if eightfoot is using redis)

**Comment by karlhungus** [comment]
@ibawt shall i just make a commit with the boxen version not including the sqlite3 (maybe move the db definitions ~~off~~ over to mysql, or add `config/database.yml` to .gitignore and create a `config/database-example.yml` and update readme?

I'm kind of partial to it more working out of the box for people who are going to participate, if you start seeing others come on board that wouldn't have `BOXEN` defined than move to the database-example.yml model

**Comment by ibawt** [comment]
yes out of the box.  Then lets just copy the shopify config.   I didn't move any test stuff as I haven't written any tests ( oh the horror. ) 

If it gets popular outside of shopify ( and pigs start flying ) we can worry about a better solution for others.

Do you want to remove teh references to sqllite and then :shipit:

**Pull request merged**